### PR TITLE
[Doc Reorg] Cheat Sheet

### PR DESCRIPTION
**Pull request by Br4nd0R**
* This was adapted from the internal confluence article "CumulusCI At a Glance".
* Most of the content is the same. I removed references talking about "team" related items including the section that covers trial orgs.
* I've created another user story to update the cheat sheet to be persona based after we release the initial beta version of the docs.

View the rendered version of the cheatsheet [here](https://cumulusci.readthedocs.io/en/feature-doc-reorg__cheatsheet/cheat_sheet.html).

### CODE REVIEW

File: docs/cheat_sheet.rst [hunk 1]
```diff
@@ -1,4 +1,221 @@
 Cheat sheet
 ===========
+CumulusCI offers a great deal of functionality out of the box.
+This cheatsheet is intended to provide a very brief summary of the most
```
Comment (davidmreed) [inline docs/cheat_sheet.rst:4]:
```suggestion
This cheat sheet is intended to provide a very brief summary of the most
```

File: docs/cheat_sheet.rst [hunk 2]
```diff
@@ -1,4 +1,221 @@
...
+important commands to start working in scratch orgs using CumulusCI,
+using the basic flows and tasks supplied with the tool. 
 
-Quick reference for common commands
+Naming and Manipulating Orgs
+----------------------------
+CumulusCI supplies a collection of named org configurations by default.
+To see what org configurations are available, run cci org list.
```
Comment (davidmreed) [inline docs/cheat_sheet.rst:11]:
```suggestion
To see what org configurations are available, run ``cci org list``.
```

File: docs/cheat_sheet.rst [hunk 3]
```diff
@@ -1,4 +1,221 @@
...
+You can provide those names to any of the commands in this guide.
+Common examples include dev, qa, beta, and release.
```
Comment (davidmreed) [inline docs/cheat_sheet.rst:13]:
```suggestion
Common examples include ``dev``, ``qa``, ``beta``, and ``release``. 
```

File: docs/cheat_sheet.rst [hunk 4]
```diff
@@ -1,4 +1,221 @@
...
 
-Quick reference for common commands
+Naming and Manipulating Orgs
+----------------------------
+CumulusCI supplies a collection of named org configurations by default.
+To see what org configurations are available, run cci org list.
+You can provide those names to any of the commands in this guide.
+Common examples include dev, qa, beta, and release. 
+Org names are associated with a scratch org definition file stored in
+the project's orgs directory, which determines how the scratch org is set up.
```
Comment (davidmreed) [inline docs/cheat_sheet.rst:15]:
```suggestion
the project's ``orgs`` directory. The definition file determines how the scratch org is set up. 
```

File: docs/cheat_sheet.rst [hunk 5]
```diff
@@ -1,4 +1,221 @@
...
+
+Remove an org name
+^^^^^^^^^^^^^^^^^^
+    
+.. code-block::
+
+    $ cci org remove <org_name>
+
+Note that you will not be able to remove built-in org names, 
+but you can remove names you created with cci org scratch.
```
Comment (davidmreed) [inline docs/cheat_sheet.rst:87]:
```suggestion
but you can remove names you created with ``cci org scratch``.
```

File: docs/cheat_sheet.rst [hunk 6]
```diff
@@ -1,4 +1,221 @@
...
+
+
+
+Connect to a persistent org (sandbox, developer edition)
```
Comment (davidmreed) [inline docs/cheat_sheet.rst:91]:
```suggestion
Connect to a persistent org (sandbox, Developer Edition)
```

File: docs/cheat_sheet.rst [hunk 7]
```diff
@@ -1,4 +1,221 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: console
+
+    $ cci org connect <org_name> 
+        
+Use the ``--sandbox`` if this is a sandbox.
```
Comment (davidmreed) [inline docs/cheat_sheet.rst:98]:
```suggestion
Use the ``--sandbox`` option if this is a sandbox or any org that uses the ``test.salesforce.com`` login endpoint.
```

File: docs/cheat_sheet.rst [hunk 8]
```diff
@@ -1,4 +1,221 @@
...
+
+Building Orgs
+-------------
+Every CumulusCI project includes one or more flows that build an org for a specific purpose or workflow.
+These flows may be customized for the project, or may be unique to the project.
+Below are a collection of the standard org building flows that you should expect to find in any CumulusCI project.
+
+.. note::
+
+    This section relies on concepts introduced in :ref:`key concepts` section of the documentation.
```
Comment (davidmreed) [inline docs/cheat_sheet.rst:111]:
```suggestion
    This section relies on concepts introduced in the :ref:`key concepts` section of the documentation.
```

File: docs/cheat_sheet.rst [hunk 9]
```diff
@@ -1,4 +1,221 @@
...
+    or allowed to run against a configured default org.
+
+
+
+Flows for Building Orgs
+^^^^^^^^^^^^^^^^^^^^^^^
+
+``qa_org``
+********** 
+This builds an unmanaged org that is designed for QA use. Should be used with an org whose configuration is ``qa``.
```
Comment (davidmreed) [inline docs/cheat_sheet.rst:126]: Let's replace all the "this"es with "This flow".

File: docs/cheat_sheet.rst [hunk 10]
```diff
@@ -1,4 +1,221 @@
...
+``regression_org``
+******************
+This builds a managed org that starts with the latest release installed and is then upgraded to the latest beta to simulate a subscriber upgrade. Should be used with an org whose configuration is ``release``
+
+Your project may provide additional org building flows.
+Consult the project's automation documentation for more details.
+
+.. caution::
+
+    We do not recommend running an org-building-flow against the same scratch org multiple times.
```
Comment (davidmreed) [inline docs/cheat_sheet.rst:149]:
```suggestion
    We do not recommend running an org-building flow against the same scratch org multiple times.
```

File: docs/cheat_sheet.rst [hunk 11]
```diff
@@ -1,4 +1,221 @@
...
+
+
+Execute Apex unit tests in an org
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: console
+
+    $ cci task run run_tests
+
+Use ``--managed True`` if the org is managed.
```
Comment (davidmreed) [inline docs/cheat_sheet.rst:188]: This is no longer required - CCI will infer it.

File: docs/cheat_sheet.rst [hunk 12]
```diff
@@ -1,4 +1,221 @@
...
+
+Retrieve changes to local repository
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block::
+
+    $ cci task run retrieve_changes
+
+
+More Resources
```
Comment (davidmreed) [inline docs/cheat_sheet.rst:218]: We should probably just remove this section for the documentation.

**Review by davidmreed — CHANGES_REQUESTED** [review]

**Review by davisagli — COMMENTED** [review]
I'd be interested in other perspectives, but at first glance, this is too verbose for a cheat sheet. I was thinking of something in tabular format, with a goal on the left and the related command on the right. This is not the place for explanations; it's a place to go to remind yourself of a command that you already know how to use.

**Comment by davidmreed** [comment]
"Cheat sheet" is probably a misnomer. My conception for this resource (as originally written in Confluence) was an "at a glance" usage guide to basic commands intended for newer users. I think that has value, but see your point that it really _isn't_ a cheat sheet as such.

I don't think I have a really strong sense of what the users of this documentation set are most likely to need. The at-a-glance guide is responsive to a number of internal documentation pieces developed over the years, particularly by newer or infrequent CumulusCI users.

**Comment by Br4nd0R** [comment]
We also have a work item slated for post-beta release to revamp this with more of a persona based approach. I definitely see your point @davisagli . Would it make sense to turn this into a "quick start" section? I'm inclined to slap this on the list of things to address after the beta release.

I feel like at some point it would behoove us to get some end user feedback as well.

**Comment by davisagli** [comment]
@davidmreed @Br4nd0R Hmm, a "quick start" is different from what I was thinking -- but I certainly am not sure what is the most needed and agree that we should be guided by what users need. I'm a little worried that if we need a quick start, then we didn't do a good enough job of explaining things simply in the main documentation topics -- but again it's this challenge of how do we introduce things to new users without overwhelming, while also covering things in more depth for advanced users.

**Comment by davidmreed** [comment]
I think one reason pulling this together is valuable is that we cover scratch org environment management and task and flow runs in two separate (large) sections. Those are the key players for everyday CumulusCI use.

Since none of us have a clear sense of what a user needs, why don't we ask some users? I bet Nick L. and Michael K. would be willing to offer feedback.

**Pull request merged**